### PR TITLE
Add GitHub Actions CI support

### PR DESCRIPTION
**Pull request by maxloeffler**
### Prerequisites

- [x] I adhere to the coding conventions (described [here](../CONTRIBUTING.md)) in my code.
- [x] I have updated the copyright headers of the files I have modified.
- [x] I have written appropriate commit messages, i.e., I have recorded the goal, the need, the needed changes, and the location of my code modifications for each commit. This includes also, e.g., referencing to relevant issues.
- [x] I have put signed-off tags in *all* commits.
- [x] I have updated the changelog file [NEWS.md](../NEWS.md) appropriately.
- [x] I have checked whether I need to adjust the showcase file `showcase.R` with respect to my changes.
- [x] The pull request is opened against the branch `dev`.

### Description

Add basic GitHub Actions CI script. It is not running perfectly, sometimes it yields different results on different runners and dependency caching seems to be not working properly all the time. This is an early draft.

### Changelog

Introduce 'pull_request.yaml'.

### CODE REVIEW

File: .github/workflows/pull_request.yml [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+  build:
+    name: Build
+    
+    # change to 'runs-on: self-hosted' to run on self-hosted runners (https://docs.github.com/en/actions/using-jobs/choosing-the-runner-for-a-job)
+    runs-on: ubuntu-latest
+    
+    strategy:
+      fail-fast: false
+      matrix:
+        r-version: ['3.6', '4.0', '4.1', '4.2']
```
Comment (hechtlC) [inline .github/workflows/pull_request.yml:35]: Did you check, whether we could add the 'r-latest' here as we had it in the drone config?

**Review by hechtlC — COMMENTED** [review]
Looks good, thanks @MaLoefUDS !
I just have one comment (see inline) and once this is answered/done I think we can merge this.

File: .github/workflows/pull_request.yml (hunk 1 above)
Comment (MaLoefUDS) [reply .github/workflows/pull_request.yml:35]: I have checked it just now, the `setup-r` action provided by `r-lib/actions` allows for the string `'latest'` or `'release'` to obtain the latest release build of R. I will add it to the configuration file 😄.

File: .github/workflows/pull_request.yml [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+        uses: r-lib/actions/setup-r@v2
+        with:
+          r-version: ${{ matrix.r-version }}
+
+      - name: Cache dependencies
+        id: cache-dependencies
+        uses: pat-s/always-upload-cache@v3.0.11
+        with:
+          path: packrat
+          key: ${{ runner.os }}-deps-${{ hashFiles('**/install.R') }}
```
Comment (bockthom) [inline .github/workflows/pull_request.yml:61]: As this seems to be not working at all, please remove it.

**Review by bockthom — CHANGES_REQUESTED** [review]
The new configuration of the CI looks good to me. I only have a few comments, please see them below.

In addition, we currently have some "build status" badges for branches `dev` and `master` in our CONTRIBUTIONS.md (just search for build status in there). Could you please check if there are such build status badges also available for GitHub actions?

**Comment by bockthom** [comment]
Thanks for your adjustments, the new CI pipeline looks good now! 
Regarding the status badges, we need to wait until the new CI is in action to see whether it works or not, but it also looks good to me.

However, the CI pipeline is failing now (particularly for R3.6). 
I checked the logs of the failing tests and identified the reason for the failing tests:
There is a tiny bug in `util-read.R`, that was present for more than a year now, but was not triggered yet. However, now it is triggered because of the changes of PR #235, which we have merged to `dev` yesterday.
I have written a patch for that, so please download an apply the following patch (using `git am`):
[0001-Fix-column-access-in-remove-deleted-and-empty-user.patch](https://github.com/se-sic/coronet/files/10519921/0001-Fix-column-access-in-remove-deleted-and-empty-user.patch.txt)

In addition, please also rebase your current working branch onto the current status of `se-sic/dev`. (In case of questions, don't hesitate to ask.)

**Review by bockthom — APPROVED** [review]

**Pull request merged**